File: app/src/main/java/iuliiaponomareva/eventum/async/NewsService.java
package iuliiaponomareva.eventum.async;

import android.app.IntentService;
import android.content.Context;
import android.content.Intent;
import android.support.v4.content.LocalBroadcastManager;

import java.util.Set;

import iuliiaponomareva.eventum.data.News;
import iuliiaponomareva.eventum.util.RSSAndAtomParser;

public class NewsService extends IntentService {
    private static final String ACTION_LOAD_NEWS = "iuliiaponomareva.eventum.action.LOAD_NEWS";
    public static final String URLS = "iuliiaponomareva.eventum.extra.URLS";
    public static final String ACTION_BROADCAST_NEWS =
            "iuliiaponomareva.eventum.action.BROADCAST_NEWS";

    public NewsService() {
        super("NewsService");
    }

    public static void startActionLoadNews(Context context, String[] urls) {
        Intent intent = new Intent(context, NewsService.class);
        intent.setAction(ACTION_LOAD_NEWS);
        intent.putExtra(URLS, urls);
        context.startService(intent);
    }


    @Override
    protected void onHandleIntent(Intent intent) {
        if (intent != null) {
            final String action = intent.getAction();
            if (ACTION_LOAD_NEWS.equals(action)) {
                final String[] urls = intent.getStringArrayExtra(URLS);
                parseNews(urls);
            }
        }
    }

    private void parseNews(String[] urls) {
        RSSAndAtomParser parser = new RSSAndAtomParser();
        Intent intent = new Intent(ACTION_BROADCAST_NEWS);
        for (String url : urls) {
            Set<News> news = parser.parseNews(url);
            intent.putExtra(url, news.toArray(new News[0]));
        }
        intent.putExtra(URLS, urls);
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
    }
}

File: app/src/main/java/iuliiaponomareva/eventum/async/NewsService.kt
package iuliiaponomareva.eventum.async

import android.app.IntentService
import android.content.Context
import android.content.Intent
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import iuliiaponomareva.eventum.util.RSSAndAtomParser

class NewsService : IntentService("NewsService") {
    override fun onHandleIntent(intent: Intent?) {
        if (intent != null) {
            val action = intent.action
            if (ACTION_LOAD_NEWS == action) {
                val urls =
                    intent.getStringArrayExtra(URLS)
                parseNews(urls!!)
            }
        }
    }

    private fun parseNews(urls: Array<String>) {
        val parser = RSSAndAtomParser()
        val intent = Intent(ACTION_BROADCAST_NEWS)
        for (url in urls) {
            val news = parser.parseNews(url)
            intent.putExtra(url, news.toTypedArray())
        }
        intent.putExtra(URLS, urls)
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent)
    }

    companion object {
        private const val ACTION_LOAD_NEWS = "iuliiaponomareva.eventum.action.LOAD_NEWS"
        const val URLS = "iuliiaponomareva.eventum.extra.URLS"
        const val ACTION_BROADCAST_NEWS = "iuliiaponomareva.eventum.action.BROADCAST_NEWS"
        fun startActionLoadNews(
            context: Context,
            urls: Array<String>
        ) {
            val intent = Intent(context, NewsService::class.java)
            intent.action = ACTION_LOAD_NEWS
            intent.putExtra(URLS, urls)
            context.startService(intent)
        }
    }
}